platform/mds/mds/src/main/java/org/motechproject/mds/builder/impl/EntityMetadataBuilderImpl.java

```diff
@@ -134,8 +139,10 @@ public void addHelperClassMetadata(JDOMetadata jdoMetadata, ClassData classData,
         cmd.setIdentityType(IdentityType.APPLICATION);
         cmd.setPersistenceModifier(ClassPersistenceModifier.PERSISTENCE_CAPABLE);
 
-        InheritanceMetadata imd = cmd.newInheritanceMetadata();
-        imd.setCustomStrategy("complete-table");
+        if(entityType == EntityType.STANDARD)
+            fixDiscriminator(definition, cmd, 1);
+        // TODO: entityType == EntityType.STANDARD: did trash, and history need inheritanceMetadata?
+        // TODO: do trash and history need discriminatorMetadata?
```

Comment:
Don't have an anwser to this since I'm still struggling with the concept, but they should retain all values, so if we are adding fields from the parent to an entity, then they should retain those values.